c7n/resources/code.py

```diff
@@ -265,3 +268,149 @@ def process(self, resources):
                 self.manager.retry(client.delete_pipeline, name=r['name'])
             except client.exceptions.PipelineNotFoundException:
                 continue
+
+
+@resources.register('codedeploy-application')
+class CodeDeployApplication(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'codedeploy'
+        enum_spec = ('list_applications', 'applications', None)
+        batch_detail_spec = (
+            'batch_get_applications', 'applicationNames',
+            None, 'applicationsInfo', None)
+        id = 'applicationId'
+        name = 'applicationName'
+        date = 'createTime'
+        arn_type = "application"
+        cfn_type = "AWS::CodeDeploy::Application"
+        universal_taggable = True
+
+    def augment(self, resources):
+        resources = super().augment(resources)
+        client = local_session(self.session_factory).client('codedeploy')
+        for r in resources:
+            arn = self.generate_arn(r['applicationName'])
+            r['Tags'] = client.list_tags_for_resource(
+                ResourceArn=arn).get('Tags', [])
+        return resources
+
+    def get_arns(self, resources):
+        arns = []
+        for r in resources:
+            arns.append(self.generate_arn(r['applicationName']))
+        return arns
+
+    @property
+    def generate_arn(self):
+        if self._generate_arn is None:
```

Comment:
why did this method need to be overriden?

Reply:
i went ahead and updated the pr on this, i did note the need for name for id and different arn separate as driving some of this.

Reply:
Thanks @kapilt, looks a lot cleaner